File: src/Domain/Entities/Employee.cs
using Domain.Common;
using Domain.Enums;

namespace Domain.Entities;

public class Employee : BaseEntity
{
    public string FirstName { get; set; } = string.Empty;
    public string LastName { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string PhoneNumber { get; set; } = string.Empty;
    public EmployeeRole Role { get; set; }
    public DateTime HireDate { get; set; }
    public decimal Salary { get; set; }
    public bool IsActive { get; set; } = true;
}

File: employeeapp-ui/src/types/employee.ts
export enum EmployeeRole {
  Employee = 0,
  TeamLead = 1,
  Manager = 2,
  Director = 3
}

export interface Employee {
  id: number;
  firstName: string;
  lastName: string;
  email: string;
  phoneNumber: string;
  role: EmployeeRole;
  hireDate: string;
  salary: number;
  isActive: boolean;
  createdAt: string;
  updatedAt: string;
}

export interface CreateEmployeeRequest {
  firstName: string;
  lastName: string;
  email: string;
  phoneNumber: string;
  role: EmployeeRole;
  hireDate: string;
  salary: number;
}

export interface UpdateEmployeeRequest extends CreateEmployeeRequest {
  id: number;
  isActive: boolean;
}

export interface ApiError {
  message: string;
  statusCode: number;
}